src/components/text-field.js

```diff
@@ -38,98 +56,257 @@ const _textFieldClasses = ({ container, inputSize, success, error }) => ({
+    'hover:border-secondary-600': !focus,
+    'border-b border-interactive-500': focus && !error,
+    'border-b border-error-500': focus && error,
+    'pointer-events-none bg-secondary-100 text-secondary-400 border-secondary-400': disabled,
+  }),
+  input: clsx(`${classes.input} outline-none mb-0.5 bg-transparent font-normal tracking-md text-xs ${styles.borderlessInput}`, {
+    'text-secondary-900': isEdited,
+    'text-secondary-600': !isEdited,
+  }),
 })
 
-const TextField  = ({ classes, size, inputProps, label, maxLength, helperText, success, error, required, disabled, deleteButton, onChange, onClick, onDelete, onSubmit, ...rest }) => {
+const renderLabel = ({ label, required, textFieldClasses }) => (
+  <div className='flex flex-row'>
+    {label && <p className={textFieldClasses.label}>{label}</p>}
+    {required && <span className='flex flex-row ml-5px text-error-500'>*</span>}
+  </div>
+)
+const renderFooter = ({ helperText, maxLength, value, textFieldClasses }) => (
+  <div className={`flex flex-row ${helperText ? 'justify-between' : 'justify-end'}`}>
+    {helperText && <p className={textFieldClasses.helperText}>{helperText}</p>}
+    {maxLength && <p className={textFieldClasses.wordCount}>{value.length || 0}/{maxLength}</p>}
+  </div>
+)
+
+const TextField  = ({
+  classes, variant, size, inputProps, label, maxLength, helperText, success, error,
+  required, disabled, deleteButton, onChange, onClick, onDelete, onSubmit, linkedFields,
+  ...rest
+}) => {
   const [filled, setFilled] = useState(false)
   const [value, setValue] = useState(false)
   const [focus, setFocus] = useState(false)
+  const [isEdited, setIsEdited] = useState(false)
+  const [linkedValues, setLinkedValues] = useState(new Array(linkedFields).fill(''))
+  const [linkedIncompleteError, setLinkedIncompleteError] = useState(false)
+
+  const inputID = counter()
   const { root, input, container } = classes
-  const inputSize = _inputSize({ size })
-  const textFieldClasses = _textFieldClasses({ container, inputSize, success, error })
-  const inputBaseClasses = _inputBaseClasses({ label, inputSize, focus, success, error, root, input, filled, disabled })
+  const inputSize = _inputSize({ size, variant })
+  const textFieldClasses = _textFieldClasses({ container, inputSize, success, error: error || linkedIncompleteError, linkedFields })
+  const generateVariants = ({ linked }) => ({
+    default: _inputBaseClasses({ label, inputSize, focus, success, error: error || linkedIncompleteError, root, input, filled, disabled, linked }),
+    borderless: _borderlessClasses({ classes, isEdited, error, focus, disabled }),
+  })
 
-  const handleChange = (val) => {
-    setValue(val)
+  const handleLinkedChange = (e, i, inputID) => {
+    if ((i + 1) === linkedFields && linkedValues[i]) {
+      return onChange(linkedValues.join(''))
+    }
+
+    if (e.target.value.length > 1) {
+      if (!linkedValues.filter((r) => r).length) {
```

Comment:
when `e.target.value` length is greater than 1, wouldn't `linkedValues` length be always greater than 0? Just wondering in which cases the condition is met 🤔

Reply:
`e.target.value` length is greater than 1 would likely be when users copy/paste into single input field, the check for when `linkedValues` length is to set the values pasted only when `linkedValues` does not already have any values, otherwise pasting more than 1 character into the field is forbidden

...these user flows are purely by my assumptions tho 😅 , maybe you and @sallkall @vlatawiec @YizhiCatherineZhang have a more user-friendly flow in mind, happy to revise

Reply:
Oh I see, like how we copy-paste the verification code from the auth email 👍  wasn't aware of that use case. 
I think this can serve another case when the codes are already typed which overrides the previous ones. What do you think?

Reply:
instead of overriding the previous values (might be confusing), i filled the avail remaining boxes with the pasted value, lemme know what you think